support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/dao/NoSqlTicketRegistry.java

```diff
@@ -0,0 +1,113 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NoSqlTicketRegistry.class);
+    private static final String TICKET_GRANTING_TICKET_PREFIX = "TGT";
+    private static final String SERVICE_TICKET_PREFIX = "ST";
+
+    private NoSqlTicketRegistryDao ticketRegistryDao;
+    private LogoutManager logoutManager;
+
+    @Autowired
+    public NoSqlTicketRegistry(@Qualifier("cassandraDao") final NoSqlTicketRegistryDao ticketRegistryDao, final LogoutManager logoutManager) {
+        this.ticketRegistryDao = ticketRegistryDao;
+        this.logoutManager = logoutManager;
+    }
+
+    @Override
+    public void addTicket(final Ticket ticket) {
+        final String ticketId = ticket.getId();
+        LOGGER.debug("INSERTING TICKET {}", ticketId);
+        if (isTgt(ticketId)) {
+            ticketRegistryDao.addTicketGrantingTicket(ticket);
+        } else if (isSt(ticketId)) {
+            ticketRegistryDao.addServiceTicket(ticket);
+        } else {
+            LOGGER.error("inserting unknown ticket type {}", ticket.getClass().getName());
+        }
+    }
+
+    @Override
+    public int deleteTicket(final String id) {
+        if (deleteSingleTicket(id)) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public boolean deleteSingleTicket(final String id) {
+        LOGGER.debug("DELETING TICKET {}", id);
+        if (isTgt(id)) {
+            return ticketRegistryDao.deleteTicketGrantingTicket(id);
+        } else if (isSt(id)) {
+            return ticketRegistryDao.deleteServiceTicket(id);
+        } else {
+            LOGGER.error("deleting unknown ticket type {}", id);
+            return false;
+        }
+    }
+
+    @Override
+    public Ticket getTicket(final String id) {
+        LOGGER.debug("QUERYING TICKET {}", id);
+        if (isTgt(id)) {
+            return ticketRegistryDao.getTicketGrantingTicket(id);
+        } else if (isSt(id)) {
+            return ticketRegistryDao.getServiceTicket(id);
+        } else {
+            LOGGER.error("requesting unknown ticket type {}", id);
+            return null;
+        }
+    }
+
+    @Override
+    public void updateTicket(final Ticket ticket) {
+        final String ticketId = ticket.getId();
+        LOGGER.debug("UPDATING TICKET {}", ticketId);
+        if (isTgt(ticketId)) {
+            ticketRegistryDao.updateTicketGrantingTicket(ticket);
+        } else if (isSt(ticketId)) {
+            ticketRegistryDao.updateServiceTicket(ticket);
+        } else {
+            LOGGER.error("inserting unknown ticket type {}", ticket.getClass().getName());
+        }
+    }
+
+    @Override
+    public Collection<Ticket> getTickets() {
+        return null;
+    }
+
+    @Scheduled(initialDelayString = "${cas.ticket.registry.cleaner.startDelay:20000}",
+            fixedDelayString = "${cas.ticket.registry.cleaner.repeatInterval:60000}")
+    @Override
+    public void clean() {
```

Comment:
You don't need this method here. The default cleaner handles all of this. If you wish to control the cleaning behavior, please do that either in a separate component or a separate PR.